Cognitive-Library/SentimentAnalyzer/Sentiments.cs

```diff
@@ -9,26 +9,47 @@
 
 namespace SentimentAnalyzer
 {
-    public class Sentiments
+    /// <summary>
+    /// Thread-safe singleton to manage initialization of ML training model
+    /// to improve runtime performance.
+    /// </summary>
+    public sealed class Sentiments
     {
-        public static SentimentPrediction Predict(string text)
+        private static readonly Sentiments instance = new Sentiments();
+        private readonly PredictionEngine<Sentiment, SentimentPrediction> predictionEngine;
+
+        /// <summary>
+        /// Explicit static constructor to tell C# compiler
+        /// not to mark type as beforefieldinit
+        /// </summary>
+        static Sentiments()
         {
-            //Create MLContext to be shared across the model creation workflow objects 
-            //Set a random seed for repeatable/deterministic results across multiple trainings.
-            var assembly = Assembly.GetExecutingAssembly();
-            var stream = assembly.GetManifestResourceStream("SentimentAnalyzer.MLModels.SentimentModel.zip");
+        }
 
-            var mlContext = new MLContext(seed: 1);
-            Sentiment statement = new Sentiment { Col0 = text };
-            ITransformer trainedModel = mlContext.Model.Load(stream, out var modelInputSchema);
+        private Sentiments()
+        {
+            var assembly = Assembly.GetExecutingAssembly();
+            using (var stream = assembly.GetManifestResourceStream("SentimentAnalyzer.MLModels.SentimentModel.zip"))
+            {
+                var mlContext = new MLContext();
```

Comment:
According to docs for `MLContext`, `seed` is not used on an already-trained model so I left it out.

Reply:
Makes sense.